docs/en/deployment-guides/horizontal-scaling.md

```diff
@@ -0,0 +1,468 @@
+
+    <coordination_settings>
+        <operation_timeout_ms>10000</operation_timeout_ms>
+        <session_timeout_ms>30000</session_timeout_ms>
+        <raft_logs_level>trace</raft_logs_level>
+    </coordination_settings>
+
+    <raft_configuration>
+    # highlight-start
+        <server>
+            <id>1</id>
+            <hostname>chnode1</hostname>
+            <port>9444</port>
+        </server>
+    # highlight-end
+        <server>
+            <id>2</id>
+            <hostname>chnode2</hostname>
+            <port>9444</port>
+        </server>
+        <server>
+            <id>3</id>
+            <hostname>chnode3</hostname>
+            <port>9444</port>
+        </server>
+    </raft_configuration>
+  </keeper_server>
+</clickhouse>
+```
+
+### Macros configuration
+
+The macros `shard` and `replica` reduce the complexity of distributed DDL.  The values configured are automatically substituted in your DDL queries, which simplifies your DDL.  The macros for this configuration specify the shard and replica number for each node.  
+In this 2 shard 1 replica example, the replica macro is `replica_1` on both chnode1 and chnode2 as there is only one replica.  The shard macro is `1` on chnode1 and `2` on chnode2. 
+
+```xml title="macros.xml on chnode1"
+<clickhouse>
+  <macros>
+ # highlight-next-line
+    <shard>1</shard>
+    <replica>replica_1</replica>
+  </macros>
+</clickhouse>
+```
+
+### Replication and sharding configuration
+
+Starting from the top:
+- The remote_servers section of the XML specifies each of the clusters in the environment. The attribute `replace=true` replaces the sample remote_servers in the default ClickHouse configuration with the remote_server configuration specified in this file.  Without this attribute the remote servers in this file would be appended to the list of samples in the default.  
+- In this example, there is one cluster named `cluster_2S_1R`.
+- A secret is created for the cluster named `cluster_2S_1R` with the value `mysecretphrase`.  The secret is shared across all of the remote servers in the environment to ensure that the correct servers are joined together.
+- The cluster `cluster_2S_1R` has two shards, and each of those shards has one replica.  Take a look at the architecture diagram toward the beginning of this document, and compare it with the two `shard` definitions in the XML below.  In each of the shard definitions there is one replica.  The replica is for that specific shard.  The host and port for that replica is specified.  The replica for the first shard in the configuration is stored on `chnode1`, and the replica for the second shard in the configuration is stored on `chnode2`.
+- Internal replication for the shards is set to true.  Each shard can have the internal_replication parameter defined in the config file. If this parameter is set to true, the write operation selects the first healthy replica and writes data to it.
```

Comment:
I will add the link to that page at the bottom of the guide; I am trying to avoid having many links in the main body of tutorials as (hopefully) people should be able to succeed by following the steps on the page.